src/event_handler.rs

```diff
@@ -192,22 +188,125 @@ async fn send_execution_output(app_state: Data<AppState>, mut rx: Receiver<JobRe
+                    gas_price.unwrap_err()
                 );
+                sleep(Duration::from_millis(10)).await;
+                continue;
+            };
 
-                send_txn(txn).await
-            },
-            should_retry,
-        )
-        .await;
+            // If required retrieve the current nonce from the network and retry otherwise
+            if update_nonce == true {
+                let new_nonce_to_send = http_rpc_client
+                    .get_transaction_count(http_rpc_client.address(), None)
+                    .await;
+                if new_nonce_to_send.is_err() {
+                    eprintln!(
+                        "Failed to fetch current nonce for the gas address ({:?}): {:?}",
+                        http_rpc_client.address(),
+                        new_nonce_to_send.unwrap_err()
+                    );
+                    continue;
+                };
+                nonce_to_send = new_nonce_to_send.unwrap();
+                update_nonce = false;
+            }
 
-        let Ok(_) = txn_result else {
-            eprintln!(
-                "Failed to submit the execution output transaction for job id {}: {:?}",
-                job_output.id,
-                txn_result.unwrap_err()
+            // Update metadata to be used for sending the transaction and send it to the common chain
+            let txn = txn
+                .gas(estimated_gas + GAS_LIMIT_BUFFER)
+                .nonce(nonce_to_send)
+                .gas_price(gas_price);
+            let pending_txn = txn.send().await;
+            let Ok(pending_txn) = pending_txn else {
+                let error_string = format!("{:?}", pending_txn.unwrap_err());
+                eprintln!(
+                    "Failed to send the execution {} transaction for job id {}: {}",
+                    txn_type, job_id, error_string
+                );
+
+                // Retry if 'nonce too low' error encountered
+                if error_string.contains("code: -32000") && error_string.contains("nonce") {
+                    update_nonce = true;
+                    continue;
+                }
+
+                // Retry after a delay if connection failed
+                if error_string.contains("code: -32000") && error_string.contains("connection") {
+                    sleep(Duration::from_millis(100)).await;
+                    continue;
+                }
+
+                break;
+            };
+
+            let pending_tx_hash = pending_txn.tx_hash();
+            println!(
+                "Execution {} transaction successfully sent for job id {} with nonce {} and hash {:?}",
+                txn_type, job_id, nonce_to_send, pending_tx_hash
             );
-            continue;
-        };
+
+            // Monitor the transaction sent for block confirmation
+            let app_state_clone = app_state.clone();
+            tokio::spawn(async move {
```

Comment:
do we really need this thread for each txn? Can improve if there is only one thread watching the head of the outstanding list of txns.

Reply:
Exploring this.